### PR TITLE
Softmax should occur in f32

### PR DESCRIPTION
**Pull request by rsuderman**
If softmax is in f16 there is a risk of under/overflow during softmax. Changing to f32 makes this numerically stable.

### CODE REVIEW

File: sharktank/sharktank/models/llm/export.py [hunk 1]
```diff
@@ -76,10 +76,14 @@ def prefill(self, tokens, start_pos, seq_lens, seq_block_ids, cs):
         logits = ops.unshard(logits)
 
         if self.config.logits_normalization == "softmax":
+            logits = logits.to(dtype=torch.float32)
             logits = ops.softmax(logits, dim=-1)
```
[the remark this answers is not in the document]
Comment (rsuderman) [reply sharktank/sharktank/models/llm/export.py:80]: We can, I just usually do explicit ops. Its really just up to the user.
